### PR TITLE
Create an assembly per each LB config use case

### PR DESCRIPTION
**Pull request by asteflova**
The current way the procedures for configuring LB are documented is rather confusing -- a lot of different procedures are included within one assembly (yes Puppet/no Puppet, custom certs/default certs), and it takes quite some time to figure out which procedures are related to which use case. With this PR, I aim to create multiple assemblies where each assembly guides users through their scenario from beginning to end:

* Users can now find all related procedures and other information (prerequisites, limitations) under one assembly heading.
* The number of xrefs leading to other procedures has been reduced; there is no need for them now.
* Each assembly now clearly links to support limitations (this is to address https://bugzilla.redhat.com/show_bug.cgi?id=2251030 ).
* While I was at it, I also reworded procedure introductions to make them more descriptive and user-focused (mainly by removing self-referential wording like "This section/procedure")

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

File: guides/common/modules/con_configuring-smartproxyservers-with-default-ssl-certificates-for-load-balancing-with-puppet.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[id="configuring-{smart-proxy-context}-servers-with-default-ssl-certificates-for-load-balancing-with-puppet_{context}"]
+= Configuring {SmartProxyServers} with default SSL certificates for load balancing (with Puppet)
+
+If you use Puppet in your {Project} setup, you can configure one or more {SmartProxyServers} that use default SSL certificates for load balancing.
+
+To do this, you configure Puppet certificate signing on one of your {SmartProxyServers}.
+Then, you configure each other {SmartProxy} used for load balancing to use the certificates.
```
Comment (Lennonka) [inline guides/common/modules/con_configuring-smartproxyservers-with-default-ssl-certificates-for-load-balancing-with-puppet.adoc:7]:
```suggestion
Then, you configure each remaining Puppet {SmartProxy} used for load balancing to use the certificates.
```

File: guides/common/modules/con_configuring-smartproxyservers-with-default-ssl-certificates-for-load-balancing-with-puppet.adoc [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+The first {SmartProxyServer} will generate and sign Puppet certificates for the other {SmartProxies} configured for load balancing.
```
Comment (Lennonka) [inline guides/common/modules/con_configuring-smartproxyservers-with-default-ssl-certificates-for-load-balancing-with-puppet.adoc:8]:
```suggestion
The first {SmartProxyServer} will generate and sign Puppet certificates for the remaining {SmartProxies} configured for load balancing.
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
LGTM, just a couple of nitpicks

File: guides/common/modules/ref_prerequisites-configuring-smartproxyservers-for-load-balancing.adoc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[id="prerequisites-configuring-{smart-proxy-context}-servers-for-load-balancing_{context}"]
+== Prerequisites
+:context: {parent-context}
+:!parent-context:
```
[the remark this answers is not in the document]
Comment (asteflova) [reply guides/common/modules/ref_prerequisites-configuring-smartproxyservers-for-load-balancing.adoc]: Applied.

File: guides/common/assembly_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc [hunk 4]
```diff
@@ -0,0 +1,29 @@
+include::modules/con_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc[]
+
+:parent-context: {context}
+:context: LB-custom-ssl-with-puppet 
+include::modules/ref_prerequisites-configuring-smartproxyservers-for-load-balancing.adoc[]
+
```
Comment (Lennonka) [inline guides/common/assembly_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc:6]:
```suggestion
```
The others as well.

Comment (Lennonka) [reply guides/common/assembly_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc:6]: Btw, shouldn't this module have offset +1?

File: guides/common/modules/ref_prerequisites-configuring-smartproxyservers-for-load-balancing.adoc [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[id="prerequisites-configuring-{smart-proxy-context}-servers-for-load-balancing_{context}"]
+== Prerequisites
```
Comment (Lennonka) [inline guides/common/modules/ref_prerequisites-configuring-smartproxyservers-for-load-balancing.adoc:2]:
```suggestion
= Prerequisites
```

And leveloffset=+1 for the module inclusion.
```suggestion
== Prerequisites
```
```suggestion
== Prerequisites
```

**Review by Lennonka — CHANGES_REQUESTED** [review]

File: guides/common/assembly_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc (hunk 4 above)
Comment (asteflova) [reply guides/common/assembly_configuring-smartproxyservers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc]: @Lennonka It looks like the missing leveloffset was the culprit all along. The book builds without the blank lines now that I've fixed that.

**Comment by asteflova** [comment]
Merged to "master" and cherry-picked:

72013f0895..1a08c74dd5  3.11 -> 3.11

**Pull request merged**